netket/experimental/sampler/rules/fermion_2nd.py

```diff
@@ -0,0 +1,63 @@
+import jax
+import jax.numpy as jnp
+from netket.sampler.rules import ExchangeRule
+
+
+class FermionExchangeRule(ExchangeRule):
+    """Exchange rule for fermions.
+
+    Works similarly to exchange rule, but:
+    takes into account that only occupied orbitals
+    can be exchanged with unoccupied ones.
+    """
+
```

Comment:
I would include this in here, as it will be the standard use-case.

We can also ask the Hilbert space in here in order to make sure we get the right number of spin values... I dont know if this makes sense, but it seems to me it might?

```suggestion

    def __init__(
        self,
        hilbert, 
        *,
        clusters: Optional[list[tuple[int, int]]] = None,
        graph: Optional[AbstractGraph] = None,
        d_max: int = 1,
        copy_per_spin: bool = True,
    ):
        r"""
        Constructs the Exchange Rule.

        You can pass either a list of clusters or a netket graph object to
        determine the clusters to exchange.

        Args:
            clusters: The list of clusters that can be exchanged. This should be
                a list of 2-tuples containing two integers. Every tuple is an edge,
                or cluster of sites to be exchanged.
            graph: A graph, from which the edges determine the clusters
                that can be exchanged.
            d_max: Only valid if a graph is passed in. The maximum distance
                between two sites
            copy_per_spin: (default True) If True, the graph must encode the
                connectivity  between the first N physical sites having same spin, and
                it is replicated using :ref:`netket.graph.disjoint_union` other every
                spin subsector. This option conserves the number of fermions per
                spin subsector. 
        """
        if copy_per_spin and hilbert.n_spin_subsectors > 1:
            if graph is not None and graph.n_nodes == hilbert.n_orbitals:
                 graph = disjoint_union(*[graph] * hilbert.n_spin_subsectors)
            if clusters is not None and np.max(clusters) < hilbert.n_orbitals:
                clusters = np.concatenate(
                    [
                        clusters + i * hilbert.n_orbitals
                        for i in range(hilbert.n_spin_subsectors)
                    ]
                )
        super().__init__(clusters=clusters, graph=graph, d_max=d_max) 

```